plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/interactive/ElkGraphConstraintSerializer.xtend

```diff
@@ -0,0 +1,73 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package de.cau.cs.kieler.klighd.lsp.interactive
+
+import de.cau.cs.kieler.klighd.internal.util.KlighdInternalProperties
+import de.cau.cs.kieler.klighd.lsp.KGraphLanguageClient
+import de.cau.cs.kieler.klighd.lsp.KGraphLanguageServerExtension
+import java.io.ByteArrayOutputStream
+import java.util.List
+import java.util.Map
+import org.eclipse.elk.graph.ElkNode
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+
+/**
+ * Serializes constraint for an ELK graph by just adding the corresponding properties.
+ * 
+ * @author sdo
+ *
+ */
+class ElkGraphConstraintSerializer implements IConstraintSerializer {
+    
+    override canHandle(Object graph) {
+        return graph instanceof ElkNode
+    }
+    
+    override serializeConstraints(List<ConstraintProperty<Object>> changedNodes,
+        Object graph,
+        String uri,
+        KGraphLanguageServerExtension ls,
+        KGraphLanguageClient client
```

Comment:
This method interface seems a bit overloaded, wouldn't it work to just give this method the resource directly (making the ls extension superfluous) and letting this method return some change object (making the client superfluous)?

Reply:
You are right.

Reply:
I want that users still have the freedom to apply the edit operation as they want to do it. Hence, I only added helper methods to simplify this in case it works as I expect it.